chain/network/src/types.rs

```diff
@@ -199,7 +224,7 @@ pub struct AnnounceAccountRoute {
 
 #[derive(BorshSerialize, BorshDeserialize, Serialize, PartialEq, Eq, Clone, Debug)]
 pub enum HandshakeFailureReason {
-    ProtocolVersionMismatch(u32),
+    ProtocolVersionMismatch { version: u32, oldest_supported_version: u32 },
```

Comment:
doesn't this also change the protocol?

Reply:
Yes, I can increase PROTOCOL_VERSION without problem, but the real issue is with OLDEST_SUPPORTED_VERSION. It is true that we don't support version <=33 anymore after this change, but if we increase OLDEST_SUPPORTED_VERSION to 34 then nodes won't be able to talk. But this only affects `PeerMessage::HandshakeFailure` which is merely and informative message (it is used today to disconnect with relevant message for the user). In practice this message won't be used at all before the next protocol update (with handshakeV2). So we really don't need to update PROTOCOL_VERSION here.

Reply:
I see